python/anvill/binja.py

```diff
@@ -514,6 +515,11 @@ def get_variable_impl(self, address):
         """Given an address, return a `Variable` instance, or
         raise an `InvalidVariableException` exception."""
 
+        # raise exception if the variable has invalid address
+        seg = self._bv.get_segment_at(address)
```

Comment:
Add another (redundant) check on `get_segment_at` in the loop that tries to add referenced entities, so that we don't even enter here in the first place.